sm/tedge_agent/tests/main.rs

```diff
@@ -0,0 +1,39 @@
+#[cfg(test)]
+#[cfg(feature = "mosquitto-available")]
+mod tests {
+    use std::process::{Command, Stdio};
+    use std::{thread, time};
+
+    use assert_cmd::prelude::*;
+    use predicates::prelude::*;
+
+    #[test]
+    /// Tests that only one instance of `tedge_agent` is running.
+    /// This is done by spawning/running two instances of `tedge_agent`
+    /// expecting the first one to work and the second to fail.
+    fn tedge_agent_check_no_multiple_instances_running() -> Result<(), Box<dyn std::error::Error>> {
+        // running first `tedge_agent` binary
+        let mut agent = Command::cargo_bin(env!("CARGO_PKG_NAME"))?
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
```

Comment:
I would not remove redirect the errors here. Having the daemon output helps to understand why a test is failing.

I agree that the current output of the agent is too verbose. So not ideal.